lib/elixir/lib/version.ex

```diff
@@ -107,8 +107,18 @@ defmodule Version do
   @type t :: %__MODULE__{major: major, minor: minor, patch: patch, pre: pre, build: build}
 
   defmodule Requirement do
-    @moduledoc false
+    @moduledoc """
+    A struct that holds version requirement information.
+
+    The struct fields are private and should not be accessed.
+
+    See the "Requirements" section in the `Version` module
+    for more information.
+    """
+
     defstruct [:source, :matchspec, :compiled]
+
+    @typedoc false
```

Comment:
Links to `Version.Requirement.t` will open the module documentation. The anchor `#t:t/0` won't be find, but I think that's ok.

Reply:
We need to either do that or make the type as opaque, but that will lead to dialyzer errors.

Reply:
Other alternative is to make the structure public, and maybe hide the inner types?

```elixir
@typep source :: String.t()
@typep compiled :: boolean
@typep matchspec :: :ets.match_spec()

@type t :: %__MODULE__{source: source, matchspec: matchspec, compiled: compiled}
```

Reply:
@fertapric if we make it public, I'm afraid we're still allowing people to reply on the fields in the struct.

Reply:
I would go with `@opaque t` and let dialyzer warn (as we do in other places).